gym/envs/registration.py

```diff
@@ -621,30 +621,48 @@ def make(
-            and ("single_rgb_array" in render_modes or "rgb_array" in render_modes)
-        ):
+    # If we have access to metadata we check that "render_mode" is valid and see if the HumanRendering wrapper needs to be applied
+    if mode is not None:
+        if hasattr(env_creator, "metadata"):
+            if isinstance(env_creator.metadata, dict):
+                if "render_modes" in env_creator.metadata:
+                    render_modes = env_creator.metadata["render_modes"]
+
+                    # Apply the `HumanRendering` wrapper, if the mode=="human" but "human" not in render_modes
+                    if (
+                        mode == "human"
+                        and "human" not in render_modes
+                        and (
+                            "single_rgb_array" in render_modes
+                            or "rgb_array" in render_modes
+                        )
+                    ):
+                        logger.warn(
+                            "You are trying to use 'human' rendering for an environment that doesn't natively support it. "
+                            "The HumanRendering wrapper is being applied to your environment."
+                        )
+                        apply_human_rendering = True
+                        if "single_rgb_array" in render_modes:
+                            _kwargs["render_mode"] = "single_rgb_array"
+                        else:
+                            _kwargs["render_mode"] = "rgb_array"
+                    # else:
+                    #   we don't raise an error as the environment may have forgotten to add the "render_modes"
+                    #   if the render mode is not valid for an environment, the environment should raise an error
+                else:
+                    logger.warn(
+                        f"The environment {id} metadata does not include `render_modes`."
+                    )
+            else:
+                logger.warn(
+                    f"The environment {id} metadata is not a dictionary, actual type: {type(env_creator.metadata)}"
+                )
+        else:
             logger.warn(
-                "You are trying to use 'human' rendering for an environment that doesn't natively support it. "
-                "The HumanRendering wrapper is being applied to your environment."
-            )
-            _kwargs["render_mode"] = (
-                "single_rgb_array"
-                if "single_rgb_array" in env_creator.metadata["render_modes"]
-                else "rgb_array"
-            )
-            apply_human_rendering = True
-        elif mode is not None and mode not in render_modes:
-            raise error.Error(
-                f"Invalid render_mode provided: {mode}. Valid render_modes: None, {', '.join(render_modes)}"
+                f"The environment creator for {id} does not contain metadata, this could be due to metadata being set during initialisation. Creator: {env_creator}"
```

Comment:
IIRC this can also happen for older versions of python if we `make` an environment from a different package (because of something importlib does). Also, it can happen because `env_creator` is a factory function. I don't think we should raise a warning because it might actually trigger them for some of Gym's own environments.

Reply:
Do we ignore it then?

Reply:
I'm not sure. I am like 80% sure that for some python versions and some gym environments this will be a problem and a warning will be issued, which is kinda undesirable